src/ixdat/readers/zilien.py

```diff
@@ -162,150 +177,279 @@ def read(self, path_to_file, cls=ECMSMeasurement, name=None, **kwargs):
+            else:
+                column_series, aliases_part = self._zilien_dataset_part(
+                    series_header, column_headers_cut, data_cut
+                )
+                # update aliases
+                for standard_name, series_name in aliases_part.items():
+                    aliases[standard_name] += series_name
+
+            series += column_series
+
+        return series, aliases
+
+    def _zilien_dataset_part(self, series_header, column_headers, data):
+        """Process necessary data for a Zilien dataset part."""
+
+        count = self._metadata[f"{series_header}_{series_header}_count"]
+        names_and_units = [
+            self._form_names_and_unit(series_header, column_header)
+            for column_header in column_headers
+        ]
+
+        # Fill Mass aliases
+        aliases = defaultdict(list)
+        for series_name, _, standard_name in names_and_units:
             if standard_name:
                 aliases[standard_name].append(series_name)
 
-            # Form series kwargs and the series
+        # Create Ixdat series
+        column_series = self._create_series_objects(
+            column_headers, names_and_units, data[:count, :]
+        )
+
+        return column_series, aliases
+
+    def _biologic_dataset_part(self, column_headers, data):
+        """Process necessary data for a Biologic dataset part."""
+
+        count = self._metadata[f"{BIOLOGIC_SERIES_NAME}_{BIOLOGIC_SERIES_NAME}_count"]
+        names_and_units = [
+            self._form_names_and_unit(BIOLOGIC_SERIES_NAME, column_header)
+            for column_header in column_headers
+        ]
+
+        # Split rows according to techniques used according to experiments
+        # Combine the experiment numbers and the technique numbers
+        # in order to create unique identifiers for a successful split
+        exp_nums = data[:count, column_headers.index("experiment_number")]
+        tech_nums = data[:count, column_headers.index("technique_number")]
+        split_vector = exp_nums * 1000 + tech_nums
+        splits = self._get_biologic_splits(split_vector)
+
+        # Create Ixdat series
+        column_series = []
+        for begin, end in splits:
+            column_series += self._create_series_objects(
+                column_headers, names_and_units, data[begin:end, :]
+            )
+
+        return column_series
+
+    def _create_series_objects(self, column_headers, names_and_units, data):
+        """Create an Ixdat series objects from a given portion of a dataset."""
+
+        series_objects = []
+        time_series = None
+
+        for column_number, column_header in enumerate(column_headers):
+            # Skip meta columns in the EC-lab dataset
+            if column_header in ("experiment_number", "technique_number"):
+                continue
+
+            column_data = data[:, column_number]
+
+            # Skip holes in the EC-lab dataset
+            if all(np.isnan(column_data)):
```

Comment:
This can be pushed all down to numpy with `all` on the array: https://stackoverflow.com/a/60677862

Reply:
Nice. I seriously need to learn all the numpy magic.

Reply:
Well, the first step is probably just to assume that if you are to do a common operation on an entire array, then most likely numpy will have a optimized version for it.